packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -36,12 +37,18 @@ BuySellModalFooter.propTypes = {
 
 const BuySellModal = observer(({ table_type, selected_ad, should_show_popup, setShouldShowPopup }) => {
     const isMounted = useIsMounted();
-    const { general_store, order_store } = useStores();
+    const { buy_sell_store, general_store, order_store } = useStores();
+    const { local_currency } = buy_sell_store?.advert || {};
     const submitForm = React.useRef(() => {});
     const [error_message, setErrorMessage] = React.useState(null);
     const [is_submit_disabled, setIsSubmitDisabled] = React.useState(true);
-    const [page_footer_parent, setPageFooterParent] = React.useState(null);
-
+    const [page_footer_parent, setPageFooterParent] = React.useState(
+        <BuySellFormReceiveAmount
+            is_sell_advert={buy_sell_store.is_sell_advert}
+            local_currency={local_currency}
```

Comment:
We can use `buy_sell_store.advert.local_currency` here, so when the state is updated, it will be used here.

Reply:
regarding previous console warning, it appears only on localhost 

if we use `buy_sell_store.advert.local_currency` we will have an error `"Cannot read property 'local_currency' of undefined" `because we expect that` buy_sell_store.advert `will be undefined when we load advert list that's why we define` local_curency` like this:


` const { local_currency } = buy_sell_store?.advert || {};`

Reply:
We should only set state if all the props are available, otherwise we'd be checking for each one.

Reply:
they are available when they are needed = when we open buy/sell modal

Reply:
Then why did you get a "Cannot read property 'local_currency' of undefined" error?

Reply:
if we use buy_sell_store.advert.local_currency we will have an error "Cannot read property 'local_currency' of undefined"  but I don't use it when it's undefined

Reply:
Where are you checking if it's undefined?

Reply:
this `const { local_currency } = buy_sell_store?.advert || {}; ` helps us to avoid this error because if `advert` is `undefined` we won't even check `local_currency` prop there like ` buy_sell_store.advert.local_currency` because in this way we will get`  "Cannot read property 'local_currency' of undefined"`, instead we will just return `undefined`, but when we're on buy/sell moda`l buy_sell_store.advert` has an obj value and we use it

Reply:
optional chaining (?.)

Reply:
That can be done here also, right?

Reply:
Im not sure, I think it's better to move it in a separate variable

Reply:
Why?

Reply:
you're right, I can do this